app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
```

Comment:
You're right again :-) I tend to agree with the readability, so if you'll touch it because of blank check, just please add a comment explaining the expected setup. Something like if there's a vlanid set, it's expected both subnets should have it set to the same value.